### PR TITLE
Fix spotbugs warnings

### PR DESCRIPTION
**Pull request by rjoachim**
Fix most of the spotbugs warnings. Some warnings in tests could not be fixed and are now suppressed. In the classes touched I also fixed warnings of eclipse IDE and improved code formatting.

Following classes needed bigger changes:
- MimicNodeStore (Timer handling)
- NIDaAlgorithm (Timer handling)
- OptionalIntRejectedMessage (deprecated getMti(), introduced  getRejectMTI())
- Util.waitForPropertyChange() (improve wait/notify)
- TimeKeeper (refactoring)

### CODE REVIEW

File: src/org/openlcb/OptionalIntRejectedMessage.java [hunk 1]
```diff
@@ -23,48 +22,62 @@ public OptionalIntRejectedMessage(NodeID source, NodeID dest, int mti, int code)
     int mti;
     int code;
     
-    public int getMti() { return mti; }
+    @SuppressFBWarnings
```
Comment (pabender) [inline src/org/openlcb/OptionalIntRejectedMessage.java:25]: Which warnings are we suppressing here?

File: test/org/openlcb/EventIDTest.java [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 /**
  * @author  Bob Jacobsen   Copyright 2009
- * @version $Revision$
  */
 public class EventIDTest {
-
+    @SuppressFBWarnings
```
Comment (pabender) [inline test/org/openlcb/EventIDTest.java:12]: Again, what are we suppressing?

File: test/org/openlcb/EventIDTest.java [hunk 3]
```diff
@@ -108,6 +115,7 @@ public void testNotEqualsOtherType() {
         Assert.assertTrue(!e1.equals(e2));
     }
 
+    @SuppressFBWarnings
```
Comment (pabender) [inline test/org/openlcb/EventIDTest.java:118]: same

File: test/org/openlcb/GatewayTest.java [hunk 4]
```diff
@@ -20,8 +19,8 @@ public void testCtor() {
     @Test    
     public void testGet() {
```
Comment (pabender) [inline test/org/openlcb/GatewayTest.java:20]: This test needs to have some assertions added to it.  You removed setting of two unused variables, but what really should happen is that we should assert that some value is provided for g, g.getEastConnection and g.getWestConnection

File: test/org/openlcb/NodeIDTest.java [hunk 5]
```diff
@@ -1,52 +1,66 @@
...
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 /**
  * @author  Bob Jacobsen   Copyright 2009
- * @version $Revision$
  */
 public class NodeIDTest {
-
+    @SuppressFBWarnings
```
Comment (pabender) [inline test/org/openlcb/NodeIDTest.java:12]: Again, what are we suppressing?

File: test/org/openlcb/NodeIDTest.java [hunk 6]
```diff
@@ -110,6 +125,7 @@ public void testNotEqualsOtherType() {
         Assert.assertTrue(!e1.equals(e2));
     }
 
+    @SuppressFBWarnings
```
Comment (pabender) [inline test/org/openlcb/NodeIDTest.java:128]: same

File: test/org/openlcb/protocols/TimeKeeperTest.java [hunk 7]
```diff
@@ -1,14 +1,17 @@
...
 
-import static org.junit.Assert.*;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 /**
  * Created by bracz on 9/27/18.
  */
 public class TimeKeeperTest {
     public static class FakeTimeKeeper extends TimeKeeper {
+        @SuppressFBWarnings
```
Comment (pabender) [inline test/org/openlcb/protocols/TimeKeeperTest.java:14]: same

File: src/org/openlcb/OptionalIntRejectedMessage.java (hunk 1 above)
Comment (rjoachim) [reply src/org/openlcb/OptionalIntRejectedMessage.java]: Fixed

File: test/org/openlcb/EventIDTest.java (hunk 2 above)
Comment (rjoachim) [reply test/org/openlcb/EventIDTest.java]: Fixed

File: test/org/openlcb/EventIDTest.java (hunk 3 above)
Comment (rjoachim) [reply test/org/openlcb/EventIDTest.java]: Fixed

File: test/org/openlcb/NodeIDTest.java (hunk 5 above)
Comment (rjoachim) [reply test/org/openlcb/NodeIDTest.java]: Fixed

File: test/org/openlcb/NodeIDTest.java (hunk 6 above)
Comment (rjoachim) [reply test/org/openlcb/NodeIDTest.java]: Fixed

File: test/org/openlcb/protocols/TimeKeeperTest.java (hunk 7 above)
Comment (rjoachim) [reply test/org/openlcb/protocols/TimeKeeperTest.java]: Fixed

File: test/org/openlcb/GatewayTest.java (hunk 4 above)
Comment (rjoachim) [reply test/org/openlcb/GatewayTest.java:20]: Added assertions

**Pull request merged**